### DISCUSSION TITLE
Best option for responsive webserver?

### DISCUSSION THREAD

**Discussion by IoTrees** [micropython/micropython#12875] [RP2040 / Pico] [answered]
Does anyone have any recommendations for having a more robust or responsive webserver hosted by RPi Pico?

I am currently using the tutorial style server that responds with a block of HTML code, and responds to a few requests. 
I would ideally like to have it provide responsive feedback, automatically updating data on an interval. 

I think I can probably achieve this with what I have using some JavaScript to refresh the page every few seconds, but curious if anyone has any suggestions for something more powerful or eloquent? 

I don't know if Apache or Nginx would be overkill, since its really just one page, a handful of requests and values, maybe one form.

**Comment by mendenm** [comment, marked as answer]
look at microdot, especially microdot asyncio.

**Comment by DavesCodeMusings** [comment]
I am admittedly biased, but I use [Thimble](https://github.com/DavesCodeMusings/thimble) for all my MicroPython web serving needs. Mostly I've used it to add a REST API to my projects, but it can serve static pages as well. There's an examples folder on the GitHub site that shows how to do things.

**Comment by sophiedegran** [comment]
Yes microdot is very usefull in async mode and have websocket so you can update your html page asking server via javascript.. Miguel have done a really fantastic work. Now we are waiting for ssl in async mode which would be done in micropython V1.22 and Miguel would implemented it.

**Comment by IoTrees** [comment]
I ended up using microdot, and I am pretty pleased with it so far, thanks for all the suggestions!
